src/internal/connector/onedrive/collection.go

```diff
@@ -286,6 +300,58 @@ func (od *metadataItem) ModTime() time.Time {
 	return od.modTime
 }
 
+// getDriveItemContent fetch drive item's contents with retries
+func(oc *Collection) getDriveItemContent(
+	ctx context.Context,
+	item models.DriveItemable,
+	errs *fault.Bus,
+) (io.ReadCloser, error) {
+	var (
+		itemID   = ptr.Val(item.GetId())
+		itemName = ptr.Val(item.GetName())
+		el       = errs.Local()
+
+		itemData io.ReadCloser
+		err      error
+	)
+
+	// Initial try with url from delta + 2 retries
+	for i := 1; i <= maxRetries; i++ {
+		_, itemData, err = oc.itemReader(ctx, oc.itemClient, item)
+
+		if err == nil || !graph.IsErrUnauthorized(err) {
+			break
+		}
+
+		// Assume unauthorized requests are a sign of an expired jwt
+		// token, and that we've overrun the available window to
+		// download the actual file.  Re-downloading the item will
+		// refresh that download url.
+		di, diErr := oc.itemGetter(ctx, oc.service, oc.driveID, itemID)
+		if diErr != nil {
+			return nil, errors.Wrap(diErr, "retrieving expired item")
+		}
+		item = di
+	}
+
+	// check for errors following retries
+	if err != nil {
+		if item.GetMalware() != nil || clues.HasLabel(err, graph.LabelsMalware) {
+			logger.Ctx(ctx).With("error", err.Error(), "malware", true).Error("downloading item")
+			el.AddSkip(fault.FileSkip(fault.SkipMalware, itemID, itemName, graph.MalwareInfo(item)))
+		} else {
+			logger.Ctx(ctx).With("error", err.Error()).Error("downloading item")
+			el.AddRecoverable(clues.Stack(err).WithClues(ctx).Label(fault.LabelForceNoBackupCreation))
+		}
+
+		// return err, not el.Err(), because the lazy reader needs to communicate to
+		// the data consumer that this item is unreadable, regardless of the fault state.
+		return nil, err
+	}
```

Comment:
Should leave this in `populateItems` because if you fail from `itemGetter` above on line 331 - you do want this logic to get invoked as well

Reply:
No, I think the error that we provide there is better. Since we are only retrying for unauthorized `retrieving expired item` is a better error that this to wrap.

Reply:
The issue is that if you return on line 332 above - you will not add the error to the error bus (the logic below is skipped). 

```
			logger.Ctx(ctx).With("error", err.Error()).Error("downloading item")
			el.AddRecoverable(clues.Stack(err).WithClues(ctx).Label(fault.LabelForceNoBackupCreation))
```

Reply:
I think I'll keep in here itself to consolidate the logic, but I've reworked the logic to make sure we do properly handle it.